src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1283,7 +1283,31 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     LOG4CXX_ERROR(logger_, "Service URLs are empty! NOT sending PT to mobile!");
     return;
   }
-  SendMessageToSDK(pt_string, urls.front().url.front());
+
+  static size_t current_app = 0;
```

Comment:
@anosach-luxoft 
I agree with you that the loop condition is complex, but there are 2 conditions while the loop should iterate:
1) the default application is found
or
2) registered application with urls is found.
I think that after requested changes from AByzhynar (temporary boolean variables) it is more readable.